File: 3174_clear_digits.cpp
class Solution {
public:
    string clearDigits(string x) {
        stack<char> y;
        for(char z : x) if(isdigit(z)) { if(y.size()) y.pop(); } else y.push(z);
        string w = "";
        while(y.size()) w = y.top() + w, y.pop();
        return w;
    }
};

File: 3174_clear_digits.java
class Solution {
    public String clearDigits(String q) {
        Stack<Character> r = new Stack<>();
        for (char t : q.toCharArray()) if (Character.isDigit(t)) { if (!r.isEmpty()) r.pop(); } else r.push(t);
        StringBuilder p = new StringBuilder();
        while (!r.isEmpty()) p.insert(0, r.pop());
        return p.toString();
    }
}
